pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/commands/BaseOptions.kt

```diff
@@ -64,6 +64,18 @@ class BaseOptions : OptionGroup() {
             throw CliException(message)
           }
       }
+
+    fun RawOption.splitProps(): NullableOption<Pair<String, String>, Pair<String, String>> {
+      return convert {
+        val parts = it.split("=")
+        if (parts.size <= 1) parts[0] to "true" else parts[0] to parts[1]
+      }
+    }
+
+    fun RawOption.associateProps():
+      OptionWithValues<Map<String, String>, Pair<String, String>, Pair<String, String>> {
+      return splitProps().multiple().toMap()
+    }
```

Comment:
I'd inline at least once (if not entirely on the `associateProps` call site:
```suggestion
    fun RawOption.associateProps():
      OptionWithValues<Map<String, String>, Pair<String, String>, Pair<String, String>> {
      return convert {
        val parts = it.split("=")
        parts[0] to (parts.getOrNull(1) ?: "true")
      }.multiple().toMap()
    }
```